src/lib.rs

```diff
@@ -134,7 +135,11 @@ fn compile_machine_policies_internal(
             }
         }
 
-        compile::insert_extend_annotations(&mut type_map, extend_annotations);
+        compile::insert_extend_annotations(
+            &mut type_map,
+            extend_annotations.clone(),
```

Comment:
This seems wrong.  Now the early annotations are being drained from a clone of the map, not the map itself.  Can we modify insert_extend_annotations() to take a mutable reference to the map, rather than taking ownership?

Reply:
I have tried both a mutable and non-mutable reference here and got into all sorts of trouble.

If I do a non-mutable reference it complains that:
```
   --> src/compile.rs:256:42
    |
256 |                     t.annotations.insert(a);
    |                                   ------ ^ expected enum `internal_rep::AnnotationInfo`, found `&internal_rep::AnnotationInfo`
    |                                   |
    |                                   arguments to this function are incorrect
```
If I dereference `a` with the above I get:
```
    |
256 |                     t.annotations.insert(*a);
    |                                          ^^ move occurs because `*a` has type `internal_rep::AnnotationInfo`, which does not implement the `Copy` trait
```


If I use a mutable reference I get:
```
   --> src/compile.rs:254:22
    |
254 |             for a in annotations {
    |                      ^^^^^^^^^^^ `std::collections::BTreeSet<internal_rep::AnnotationInfo>` is not an iterator
    |
    = help: the trait `Iterator` is not implemented for `std::collections::BTreeSet<internal_rep::AnnotationInfo>`
    = help: the following other types implement trait `IntoIterator`:
              &'a std::collections::BTreeSet<T, A>
              std::collections::BTreeSet<T, A>
    = note: required for `&mut std::collections::BTreeSet<internal_rep::AnnotationInfo>` to implement `Iterator`
    = note: required for `&mut std::collections::BTreeSet<internal_rep::AnnotationInfo>` to implement `IntoIterator`
```

Reply:
I guess what we really need here is [drain_filter](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.drain_filter), but that's nightly only, and likely to remain that way for a while.

I guess the immutable reference option is our best bet.  Just use clone() instead of dereference to get it working, and add a comment on insert_extend_annotations() to convert to drain_filter() once it's stabilized.